pkg/auth/cleanup/service.go

```diff
@@ -181,6 +191,29 @@ func (s *Service) deleteUsers(config *v3.AuthConfig) error {
 	return nil
 }
 
+// deleteTokens deletes all the tokens created with the disabled provider
+func (s *Service) deleteTokens(config *v3.AuthConfig) error {
+	if config == nil {
+		return errAuthConfigNil
+	}
+
+	tokens, err := s.tokensCache.List(labels.Everything())
```

Comment:
Will this work ok when the cluster has tens of thousands of tokens?

Reply:
I guess we delete users with the same logic, so...if it works there, it'll work here?

Reply:
It's deleting _thousands_ of resources in a single sweep that would concern me.

This is done in the background, and worst case, if it times out, it will retry?

Reply:
It (authconfig controller) should retry on stratup + periodically until the the config is locked.

Reply:
@pmatseykanets UI side, what is happening? Is the UI waiting for the resources being deleted, or is it done in the background?

Maybe this is "ok" for other resources, but with the possibly thousands of tokens we should be more careful.

cc-ing @MbolotSuse and @richard-cox

Reply:
If this is the context i think it is, the UI waits for a response to a POST to the `disable` `actions` url. Not quite sure on the plumbing between there and here (if rancher responds with a 200 once request received or it does everything synchronously)

Reply:
Yes, the context is right. My understanding is that the `authconfig` is deleted synchronously, while the associated resources in the background with this controller. So we should be fine, but I wanted to double check with someone more aware of the workflow.

Reply:
It's cleaned in the background https://github.com/rancher/rancher/blob/release/v2.9/pkg/controllers/management/auth/auth_config.go#L117-L147

Reply:
> It's deleting thousands of resources in a single sweep that would concern me.
It's still a tiny fraction of `{CR|PR}TB`s thought, right?

Reply:
Yes, in this specific case, it's deleting _users_ and _tokens_ without paginating the queries.

But this is why https://github.com/rancher/rancher/pull/46250#discussion_r1686354903 we are doing it successfully just now...so...maybe not a priority for this case.